tests/model/test_rnn.py

```diff
@@ -84,9 +89,179 @@ def test_correct_rnn_type_and_size_returned(
         bias += getattr(rnn.rnn, f"bias_hh_l{l}")[
             hidden_size : 2 * hidden_size
         ]
+        bias = torch.tensor(bias).cpu()
         assert torch.allclose(bias, torch.tensor(forget_gate_bias))
 
 
+@given(
+    rnn_type=rnn_types(),
+    input_size=st.integers(min_value=1, max_value=8),
+    hidden_size=st.integers(min_value=1, max_value=8),
+    num_layers=st.integers(min_value=1, max_value=3),
+    bias=st.booleans(),
+    dropout=st.floats(min_value=0.0, max_value=1.0, allow_nan=False),
+    bidirectional=st.booleans(),
+    forget_gate_bias=st.one_of(
+        st.none(), st.floats(min_value=-10.0, max_value=10.0, allow_nan=False)
+    ),
+    batch_first=st.booleans(),
+    batch_size=st.integers(min_value=1, max_value=4),
+    max_seq_len=st.integers(min_value=1, max_value=8),
+)
+@settings(deadline=4000)
+def test_rnn_forward_pass_no_hidden(
+    rnn_type: RNNType,
+    input_size: int,
+    hidden_size: int,
+    num_layers: int,
+    bias: int,
+    dropout: float,
+    bidirectional: bool,
+    forget_gate_bias: float,
+    batch_first: bool,
+    batch_size: int,
+    max_seq_len: int,
+) -> None:
+    """Tests forward rnn pass with no hidden state passed"""
+
+    rnn = RNN(
+        rnn_type=rnn_type,
+        input_size=input_size,
+        hidden_size=hidden_size,
+        num_layers=num_layers,
+        bias=bias,
+        # warning raised when num_layers > 1 and dropout != 0.0 as this option
+        # set does not make sense:
+        dropout=0.0 if num_layers == 1 else dropout,
+        bidirectional=bidirectional,
+        forget_gate_bias=forget_gate_bias,
+        batch_first=batch_first,
+    )
+    num_directions = 1 + int(bidirectional)
+    tensor = torch.empty(
+        [max_seq_len, batch_size, input_size], requires_grad=False
+    ).normal_()
+
+    in_seq_lens = torch.randint(
+        low=1,
+        high=max_seq_len + 1,
+        size=[batch_size],
+        dtype=torch.int32,
+        requires_grad=False,
+    )
```

Comment:
Could this be moved into a Search Strategy that, for example, generates a valid tuple of `max_seq_len`, `batch_size` and `in_seq_lens`? I think this would also help with reproducibility since all the generation would be controlled by hypothesis rather than having just this done by PyTorch. It would make the code a bit uglier though so I'm happy to leave it as if you think it would be too clunky.

Reply:
Discussed in person but this will make the hypothesis testing v. slow as it would make sense to do this for all tensor inputs to the network. We should consider adding the following in future:
https://github.com/pytorch/pytorch/blob/master/caffe2/python/hypothesis_test.py